sourmash/_minhash.py

```diff
@@ -586,3 +587,58 @@ def moltype(self):                    # TODO: test in minhash tests
             return 'hp'
         else:
             return 'DNA'
+
+    def add_hash_with_abundance(self, hash, abundance):
```

Comment:
I think @ctb was referring to exposing [this Rust method](https://github.com/dib-lab/sourmash/blob/80d975c4684283cbf9ecc8ec77adb4fad9da8888/src/core/src/sketch/minhash.rs#L310) to Python, not reimplementing it here.

To expose it, you can define a new function in `src/core/src/ffi/minhash.rs` (which will be very similar to [`add_hash`](https://github.com/dib-lab/sourmash/blob/80d975c4684283cbf9ecc8ec77adb4fad9da8888/src/core/src/ffi/minhash.rs#L93) but taking an extra `abundance: u64` argument), and rerun `make include/sourmash.h` to add it to the header used by CFFI.

After all that, you can replace this method with

```suggestion
    def add_hash_with_abundance(self, hash, abundance):
        self._methodcall(lib.kmerminhash_add_hash_with_abundance, hash, abundance)
```

Reply:
And I didn't explain it very well... Maybe these two sections from the docs can help:
- https://sourmash.readthedocs.io/en/latest/developer.html#code-organization
- https://sourmash.readthedocs.io/en/latest/developer.html#exposing-new-functions-on-the-ffi
and this post I wrote some time ago (which is slightly outdated, because you don't need to update `include/sourmash.h` manually anymore):
https://blog.luizirber.org/2020/01/10/sourmash-pr/

Reply:
This was very helpful for me thank you!! If you're able to, let me know if my updated commit is more what was expected.